test/Microsoft.ML.Tests/Scenarios/Api/Metacomponents.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.FastTree;
+using Microsoft.ML.Runtime.Learners;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public partial class ApiScenariosTests
+    {
+        /// <summary>
+        /// Meta-components: Meta-components (e.g., components that themselves instantiate components) should not be booby-trapped.
+        /// When specifying what trainer OVA should use, a user will be able to specify any binary classifier.
+        /// If they specify a regression or multi-class classifier ideally that should be a compile error.
+        /// </summary>
+        [Fact]
+        void Metacomponents()
+        {
+            var dataPath = GetDataPath(IrisDataPath);
+            using (var env = new TlcEnvironment())
+            {
+                var loader = new TextLoader(env, MakeIrisTextLoaderArgs(), new MultiFileSource(dataPath));
+                var term = new TermTransform(env, loader, "Label");
+                var concat = new ConcatTransform(env, term, "Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth");
+                var trainer = new Ova(env, new Ova.Arguments
+                {
+                    PredictorType = new SubComponent<ITrainer<IPredictorProducing<float>>, SignatureBinaryClassifierTrainer>
+                     (
+                         FastTreeBinaryClassificationTrainer.LoadNameValue
+                     )
+                });
+
+                IDataView trainData = trainer.Info.WantCaching ? (IDataView)new CacheDataView(env, concat, prefetch: null) : concat;
+                var trainRoles = new RoleMappedData(trainData, label: "Label", feature: "Features");
+
+                // Auto-normalization.
+                NormalizeTransform.CreateIfNeeded(env, ref trainRoles, trainer);
+                var predictor = trainer.Train(new Runtime.TrainContext(trainRoles));
+
+                var scoreRoles = new RoleMappedData(concat, label: "Label", feature: "Features");
+                IDataScorerTransform scorer = ScoreUtils.GetScorer(predictor, scoreRoles, env, trainRoles.Schema);
+
+                // Cut out term transform from pipeline.
+                var newScorer = ApplyTransformUtils.ApplyAllTransformsToData(env, scorer, loader, term);
+                var keyToValue = new KeyToValueTransform(env, newScorer, "PredictedLabel");
+                var model = env.CreatePredictionEngine<IrisDataNoLabel, IrisPrediction>(keyToValue);
+
+                var testLoader = new TextLoader(env, MakeIrisTextLoaderArgs(), new MultiFileSource(dataPath));
+                var testData = testLoader.AsEnumerable<IrisDataNoLabel>(env, false);
+                foreach (var input in testData.Take(20))
+                {
+                    var prediction = model.Predict(input);
+                    Assert.True(prediction.PredictedLabel == "Iris-setosa");
```

Comment:
>PredictedLabel [](start = 43, length = 14)

and here you can also verify that `prediction.PredictedLabel == input.Label`

Reply:
Look on him, first he want test data without label, and then compare it label with predicted one!

---
In reply to: [208767421](https://github.com/dotnet/machinelearning/pull/653#discussion_r208767421) [](ancestors = 208767421)

Reply:
I only wanted test data without label in ONE test, not in ALL tests. The ONE test that specifically checks that label is not required :)

---
In reply to: [208767752](https://github.com/dotnet/machinelearning/pull/653#discussion_r208767752) [](ancestors = 208767752,208767421)